packages/flutter_tools/templates/app/lib/main.dart.tmpl

```diff
@@ -38,111 +38,182 @@ void main() {
-    // by the _incrementCounter method above.
+    // This method is run every time setState is called as, for example, the
+    // _incrementCounterBy method does above.
     //
-    // The Flutter framework has been optimized to make rerunning build methods
-    // fast, so that you can just rebuild anything that needs updating rather
-    // than having to individually change instances of widgets.
+    // The Flutter framework has been optimized to make build methods run
+    // quickly, so that you can rebuild anything that needs updating rather than
+    // having to individually change instances of widgets.
     return Scaffold(
       appBar: AppBar(
-        // Here we take the value from the MyHomePage object that was created by
-        // the App.build method, and use it to set our appbar title.
+        // The default background color is the primary color of the ColorScheme
+        // set on the Theme, which in this case happens to be the same white as
+        // the default Scaffold background color, so this sets the AppBar to use
+        // another color from the scheme instead, to have some contrast.
+        //
+        // Using Theme.of(context) means that this build function will
+        // automatically be called again to rebuild if the theme changes.
+        //
+        // TRY THIS: Try changing the color here to a specific color (to
+        // Colors.amber, perhaps?) and trigger a hot reload to see the AppBar
+        // change color while the other colors stay the same.
+        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
+
+        // This takes the value from the MyHomePage object that was created by
+        // the App.build method, and uses it to set the AppBar title.
         title: Text(widget.title),
       ),
-      body: Center(
-        // Center is a layout widget. It takes a single child and positions it
-        // in the middle of the parent.
+      // AnimatedContainer is similar to a Container widget, in that it defines
+      // a region that has some layout and display attributes and a child, but
+      // it also animates any changes to its parameters.
+      body: AnimatedContainer(
+        duration: const Duration(milliseconds: 200),
+        alignment: Alignment.center,
+        margin: const EdgeInsets.all(20),
+        decoration: ShapeDecoration(
+          color: Theme.of(context).colorScheme.secondaryContainer,
+          shape: StarBorder(
+            side: const BorderSide(),
+            points: _counter.toDouble(),
+            // TRY THIS: Try changing the parameters here (values from 0 to 1
+            // are allowed). When you hot reload the application, the border
+            // will animate to the new values automatically, over the duration
+            // set in the AnimatedContainer, above.
+            pointRounding: 0.4,
+            valleyRounding: 0.4,
+            innerRadiusRatio: .5,
+          ),
+        ),
```

Comment:
I'm very skeptical about adding this much here, at least unless we prove it is beneficial with substantial UI testing.